src/controller/java/src/matter/onboardingpayload/OnboardingPayload.kt

```diff
@@ -458,7 +448,7 @@ class OnboardingPayload(
    *
    * @param info Optional QR code info object to add
    */
-  private fun addOptionalExtensionData(info: OptionalQRCodeInfoExtension) {
+  fun addOptionalExtensionData(info: OptionalQRCodeInfoExtension) {
```

Comment:
This in internal implementation and should be called from 
 fun addSerialNumber(serialNumber: String)
 fun addSerialNumber(serialNumber: Int) 
 
 why remove the private accessor?

Reply:
I figured it would be more efficient to bypass the addSerialNumber since it is already being parsed out by the TLVReader. It also seems like we would need to build add methods for every extension data property defined in Onboarding Payload:

<img width="539" alt="Screenshot 2024-01-05 at 11 30 56 AM" src="https://github.com/project-chip/connectedhomeip/assets/30873477/e76b1123-e491-455a-9073-10cd3a444f7b">

Reviewing the QRCodeSetupPayloadParser.cpp file, it looks like the "addOptionalExtensionData" and "addOptionalVendorData" methods are accessible from outside of the payload class

```
        if (SetupPayload::IsCommonTag(tagNumber))
        {
            OptionalQRCodeInfoExtension info;
            info.tag = tagNumber;
            ReturnErrorOnFailure(retrieveOptionalInfo(reader, info, elemType));

            ReturnErrorOnFailure(outPayload.addOptionalExtensionData(info));
        }
        else
        {
            OptionalQRCodeInfo info;
            info.tag = tagNumber;
            ReturnErrorOnFailure(retrieveOptionalInfo(reader, info, elemType));

            ReturnErrorOnFailure(outPayload.addOptionalVendorData(info));
        }
```

Reply:
The question is if this method is an internal implementation detail of OnboardingPayload and not meant to be part of its public API?  

If we change a private method to public just for simplicity, it will increases the surface area of your API, which can make future changes more difficult. The caller has to figure out how OptionalQRCodeInfoExtension is implemented, if it is thread safe when calling fun addOptionalExtensionData.

Once a method is public, changing its signature or behavior can break existing clients of your class.

Reply:
Made this method private again and we are now calling the public addOptionalExtensionData(tag, data) method.

Reply:
thanks